samples/client/petstore/php/OpenAPIClient-php/lib/ObjectSerializer.php

```diff
@@ -262,6 +262,11 @@ public static function deserialize($data, $class, $httpHeaders = null)
             return null;
         } elseif (strcasecmp(substr($class, -2), '[]') === 0) {
             $data = is_string($data) ? json_decode($data) : $data;
+            
+            if (!is_array($data)) {
```

Comment:
I guess that it won't work in 99% cases because `json_decode` without second argument `$assoc = true` returns `StdClass` which isn't array. And my brief test confirms that:
```php
$data = '{"a":1,"b":2,"c":3,"d":4,"e":5}';
$data = is_string($data) ? json_decode($data) : $data;

if (!is_array($data)) {
    throw new \InvalidArgumentException("Invalid array '$class'");
}
// throws exception because StdClass isn't array
```

Reply:
Hi,

You are right about the fact that json_decode would return a StdClass in that scenario. Since this code is only executed when you have declared your model as an array using "[]", if such case would happen, it would be ok to send an InvalidArgumentException in my opinion. Since you have declared your model as an array and the result is an object, it's just normal to receive an InvalidArgumentException... From my understanding, there is no such things as an associative array in JSON, it's just an object. So someone declaring the value as an array with "[]" and use an object object should receive this error ?

Perhaps I'm missing something in the evaluation.

Reply:
Oh, now I see. Data should look like `["foo", "bar", "hello world"]` for `StringModel[]` class definition to pass this part of code without exception.